trieste/acquisition/utils.py

```diff
@@ -159,6 +159,29 @@ def copy_to_local_models(
     return {LocalizedTag(key, i): copy.deepcopy(global_model) for i in range(num_local_models)}
 
 
+def add_local_datasets(
+    datasets: Mapping[Tag, Dataset],
+    num_local_datasets: int,
+) -> Mapping[Tag, Dataset]:
```

Comment:
since you rely on this being a dict (or at least a mutable mapping) elsewhere:

```suggestion
) -> Dict[Tag, Dataset]:
```

Reply:
Agreed.. did consider it, but wasn't sure.